fix(replay): Fix row-clicks inside replay table

ryan953 opened the pull request:
**Purpose**
We were inconsistent with adding analytics to "view replay details" links in the rows and in the `ReplaySessionColumn`. So I wanted to simplify that and aim to have one link in the mix instead of 3 (the row, the displayName, and the shortLink).

Also, there is a problem with the existing `onRowClick` prop: it doesn't care about modifier keys. So CMD+clicking a row (to open a new browser tab) will also make the original browser tab navigate to the replay :(

**Solution**
So I spent a bunch of time, and looked at how issues does it, to make it so that rows within a `<ReplayTable>` can be clickable, and accessible at the same time.

Basically what we have now is:
- If any column is self-declared as `interactive` then we'll add the `<InteractionStateLayer>` to the row
- There is no row-level interaction that can be defined within the table... but `position: relative` was removed from the RowCell, it only exists on the Row now. 
  - So cells can be relative to the row by implementing the `:before { position: absolute; ... }` pattern.
  - If a cell does make itself absolute positioned, then z-index can control if that cell sits above, or below, it's neighbors.

There's 4 columns that each set some combination of those properties. Together these properties create a stack/layer of columns across the whole row.

| Column | `interactive: true` | `z-index: 1` | `:before{postition: absolute}` |
| --- | --- | --- | --- |
| ReplayDetailsLinkColumn | true | true | -
| ReplayPlayPauseColumn | true | true | true 
| ReplaySelectColumn | true | true | - 
| ReplaySessionColumn | true | - | true

**Layers in action**

In the **main replay table** we have the columns:
`ReplaySelectColumn | ReplaySessionColumn | ...rest`
1. The Select column uses `z-index:1` to appear on top of the Session column
2. The Session column uses the `:before` selector to make itself spread across the whole row. It's clickable everywhere, except under the select column

Something like this:
<img width="988" height="124" alt="SCR-20250716-swuc" src="https://github.com/user-attachments/assets/e03b9094-93a0-4baa-b52b-04923c94281b" />



In the **Issues>Replay** table are the columns:
`ReplayPlayPauseColumn | ReplaySessionColumn | ...rest | ReplayDetailsLinkColumn`
1. The PlayPause column uses `:before` and `z-index:1` to spread across the whole row.
2. The Session column also uses `:before` to spread across the whole row, but it's not clickable because PlayPause is above it with it's z-index
3. The Details column sets it's own z-index, which puts it above the other two columns. But it's only clickable in it's own column.

Something like this:
<img width="990" height="121" alt="SCR-20250716-swwd" src="https://github.com/user-attachments/assets/c2b55f16-4443-4fa4-9f1f-114d2b148a6f" />

ryan953 committed:
fix(replay): Remove the ability to click on replay table rows

ryan953 committed:
make groupIssues link to the replay, instead of selecting the snippet. the play button is still there for selecting

billyvg reviewed (approved):
The fake links were annoying, so v nice to have actual links. We should get the other cells figured out though as it's now inconsistent with how issues stream behaves.

ryan953 committed:
bring back full-row interaction for replay table.

adds a "View replay" button to the Issues > Replay page, because the rest of the row will select the replay to play inline

ryan953 committed:
allow any column to be interactive

ryan953 committed:
lint

ryan953 committed:
undo changes to SimpleTable

ryan953 committed:
split replayTableHeader fix to #95757

ryan953 commented on static/app/views/replays/table/replayIndexTable.tsx:
```diff
@@ -133,20 +125,6 @@ export default function ReplayIndexTable() {
           columns={columns}
           error={error}
           isPending={isPending}
-          onClickRow={({replay}) => {
-            const referrer = getRouteStringFromRoutes(routes);
-            const eventView = EventView.fromLocation(location);
-            navigate({
-              pathname: makeReplaysPathname({
-                path: `/${replay.id}/`,
-                organization,
-              }),
-              query: {
-                referrer,
-                ...eventView.generateQueryStringObject(),
-              },
-            });
-          }}
```
This is handled by `ReplaySessionColumn` now.

ryan953 commented on static/app/components/replays/table/replayTableColumns.tsx:
```diff
@@ -645,6 +724,15 @@ const CheckboxCellContainer = styled('div')`
   margin: 0 -${space(1)} 0 -${space(0.5)};
 `;
 
+const CheckboxClickTarget = styled('label')`
+  cursor: pointer;
+  display: block;
+  margin: -${p => p.theme.space.md};
+  padding: ${p => p.theme.space.md};
+  max-width: unset;
+  line-height: 0;
+`;
```
This is here to give the checkbox a little extra space to capture clicks. 

The checkbox is a small target, and missing it means a whole pageload :(

ryan953 commented on static/app/components/replays/table/replayTableColumns.tsx:
```diff
@@ -571,9 +615,18 @@ export const ReplaySlowestTransactionColumn: ReplayTableColumn = {
   },
 };
 
+const DetailsLink = styled(Link)`
+  z-index: 1;
+  margin: -${p => p.theme.space.md};
+  padding: ${p => p.theme.space.md};
+  line-height: 0;
+`;
```
This is here to give this new column a little extra invisible clickable space. 
The "View replay details" link is just an icon after all, so it's easy to miss it and click the row instead (which will load up the snippet)

ryan953 commented on static/app/components/replays/table/replayTableColumns.tsx:
```diff
@@ -494,56 +550,44 @@ export const ReplaySessionColumn: ReplayTableColumn = {
-          <Flex gap={space(0.5)} align="flex-start">
-            <DisplayNameLink
-              data-underline-on-hover
-              to={
-                isIssuesReplayList
-                  ? // if on the issues replay list, don't redirect to the details tab. this causes URL flickering
-                    {
-                      pathname: location.pathname,
-                      query: location.query,
-                    }
-                  : detailsTab()
-              }
-              onClick={trackNavigationEvent}
```
The link stuff is simplified here. 

Now, because of the z-index layering, this column doesn't need to know about the IssuesReplayList. When we're on the IssueReplayList the PlayPause column will capture all the clicks and nothing will come into here.

ryan953 committed:
more unique keys in a react loop

ryan953 committed:
fix test
